evm_loader/program/src/account/mod.rs

```diff
@@ -108,17 +104,14 @@ where
     pub const SIZE: usize = 1 + T::SIZE;
     pub const TAG: u8 = T::TAG;
 
-    pub fn from_account(
-        program_id: &Pubkey,
-        info: &'a AccountInfo<'a>,
-    ) -> Result<Self, ProgramError> {
+    pub fn from_account(program_id: &Pubkey, info: &'a AccountInfo<'a>) -> Result<Self> {
         if info.owner != program_id {
-            return Err!(ProgramError::InvalidArgument; "Account {} - expected program owned", info.key);
+            return Err(Error::AccountInvalidOwner(*info.key, *program_id));
```

Comment:
Why do you return an error without printing it? I'm afraid that `Error` will be converted into `ProgramError.Custom` value and the most interesting information will be lost. The full error message is very helpfull for debugging and understanding problems.

Reply:
In a lot of cases I don't want to print an error. It generates a lot of trash in the logs and consumes BPF.
If it is unrecoverable error that causes a transaction failure, it will be printed.